SparseGrids/tasgridWrapper.cpp

```diff
@@ -1026,13 +1059,21 @@ void TasgridWrapper::writeMatrix(const char *filename, int rows, int cols, const
     }
     ofs.close();
 }
-void TasgridWrapper::printMatrix(int rows, int cols, const double mat[]){
+void TasgridWrapper::printMatrix(int rows, int cols, const double mat[], bool isComplex){
     cout << rows << " " << cols << endl;
     cout.precision(17);
     cout << std::scientific;
     for(size_t i=0; i<((size_t) rows); i++){
-        for(size_t j=0; j<((size_t) cols); j++){
-            cout << setw(25) << mat[i* ((size_t) cols) + j] << " ";
+        if (isComplex){
+            for(size_t j=0; j<((size_t) cols); j++){
+                cout << setw(25) << mat[i* ((size_t) cols) + ((size_t) 2*j)];   // real part
+                cout << (mat[i* ((size_t) cols) + ((size_t) 2*j+1)] < 0 ? " -" : " +");
+                cout << setw(24) << abs(mat[i* ((size_t) cols) + ((size_t) 2*j+1)]) << " i   ";     // imag part
```

Comment:
* this is a bug, `abs()` works only for integer types, long, int, etc.
* you must use `fabs()` for floating point numbers
* at the end, why 2 spaces as opposed to 1?
* why not have `i` attached to the number, e.g., `1.234E-5 + 1.234.E-5i`